data/ui/preferences-device-panel.ui

```diff
@@ -564,6 +564,12 @@
                 <property name="can_focus">False</property>
                 <property name="shadow_type">none</property>
                 <child>
+                  <object class="GtkAdjustment" id="custom-battery-adjustment">
+                    <property name="upper">99</property>
+                    <property name="lower">01</property>
+                    <property name="step_increment">1</property>
+                    <property name="page_increment">5</property>
+                  </object>
```

Comment:
I think you can just set this directly actually, if that's easier:

```xml
<property name="adjustment">
  <object class="GtkAdjustment">
    <property name="upper">99</property>
    <property name="lower">01</property>
    <property name="step_increment">1</property>
    <property name="page_increment">5</property>
  </object>
</property>
```

I'm not sure if GtkBuilder allows you to place objects arbitrarily (that `<child>` tag is for the `GtkViewport`).

Reply:
I copied that placement from another repo I found at some point, they had the adjustment separate so I did the same. Your way is much simpler and less confusing.  I have implemented it locally and will push it once I have working code.

Reply:
I did not change the adjustment location as it kept giving me errors when I tried to relocate it, telling me that the adjustment wasn't allowed inside the GtkGrid. I didn't understand the error or how to fix it, so I left it as is. Its current location outside the template does not throw any errors.

Reply:
Maybe I wasn't clear with my note above, but the `<property name="adjustment">` was to be set on the spin button, such that:

```xml
<object class="GtkSpinButton">
  <property name="adjustment">
    <object class="GtkAdjustment">
      <property name="upper">99</property>
      <property name="lower">01</property>
      <property name="step_increment">1</property>
      <property name="page_increment">5</property>
    </object>
  </property>
<!-- other stuff -->
</object>
```
If you think of GtkBuilder as just objects with properties (whose value can be objects), and child objects (in `<child>` tags) it starts to become obvious how it all works :)

Outside in the template is fine also, though. I frequently put GtkDialogs in the top-level `<interface>` as well :)